src/weapons.cpp

```diff
@@ -793,7 +804,7 @@ bool WeaponDistance::useWeapon(Player* player, Item* item, Creature* target) con
 
 			for (const auto& dir : destList) {
```

Comment:
We can simplify  inside this for loop

```suggestion
			for (const auto& dir : destList) {
				const auto& tile = g_game.map.getTile(destPos.x + dir.first, destPos.y + dir.second, destPos.z);
				if (!tile) {
					continue;
				}

				if (tile->hasFlag(TILESTATE_IMMOVABLEBLOCKSOLID)) {
					continue;
				}

				if (tile->getGround()) {
					destTile = tile;
					break;
				}
			}
```